server/push_test.go

```diff
@@ -592,7 +594,131 @@ func TestShouldNotPanicIfSessionAlreadyRemoved(t *testing.T) {
 	refreshIntervalHttpPush = oldRI
 }
 
-func TestFileExtensionError(t *testing.T) {
+func TestPush_ResetWatchdog(t *testing.T) {
+
+	wgWait(&watchdogWg) // wait for any earlier tests to complete
+
+	assert := assert.New(t)
+
+	s := setupServer()
+	defer serverCleanup(s)
+
+	waitBarrier := func(ch chan struct{}) bool {
+		select {
+		case <-ch:
+			return true
+		case <-time.After(1 * time.Second):
+			return false
+		}
+	}
+
+	// override reset func with our own instrumentation
+	cancelCount := 0
+	resetCount := 0
+	var wrappedCancel func()
+	var wg sync.WaitGroup // to synchronize on cancels of the watchdog
+	timerCreationBarrier := make(chan struct{})
+	oldWatchdogReset := watchdogResetTimer
+	watchdogResetTimer = func() (context.Context, context.CancelFunc) {
+		wg.Add(1)
+		ctx, cancel := context.WithCancel(context.Background())
+		resetCount++
+		wrappedCancel = func() {
+			cancelCount++
+			cancel()
+			wg.Done()
+		}
+		timerCreationBarrier <- struct{}{}
+		return ctx, wrappedCancel
+	}
+	defer func() { watchdogResetTimer = oldWatchdogReset }()
+
+	// sanity check : normal flow should result in a single cancel
+	w := httptest.NewRecorder()
+	req := httptest.NewRequest("POST", "/live/name/0.ts", nil)
+	s.HandlePush(w, req)
+	assert.True(waitBarrier(timerCreationBarrier), "timer creation timed out")
+	wgWait(&wg)
```

Comment:
Should we assert that this returns true to make sure we didn't time out? Same comment for the rest of the `wgWait` invocations.

Reply:
Yep, fixed in https://github.com/livepeer/go-livepeer/commit/8c1e19495b48daeb15c6ab5fa9ba313d686472bc